server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -487,11 +488,72 @@ void executeRequest(
                 }
             }
         });
+        final SearchSourceBuilder source = original.source();
+        if (shouldOpenPIT(source)) {
+            openPIT(client, original, searchService.getDefaultKeepAliveInMillis(), listener.delegateFailureAndWrap((delegate, resp) -> {
```

Comment:
This honestly is the biggest question mark for me in this whole PR. No other expensive rewriting path does something like this.

We have query-rewriters that:
 - Fetch data from other indices
 - Do expensive inference calls
 - Gather metadata, etc. before querying the shards.

None of those open a PIT and then use it.

Why do we think this is substantially different?

If we think a PIT is necessary, and these retrievers rewrite into docIds & scores, why do we close it before we do the query?

If we need the docIds & scores to be the same (like we do for kNN dfs, which is ANOTHER place where we do not open a PIT), why not keep the PIT throughout the entire query phase?

Reply:
This will be leveraged only by `compound` retrievers, which for now is just `rrf` (semantic reranker will not be a compound one). As you mentioned the idea behind the PIT is to ensure that the same readers will be used for all queries throughout a rewrite phase as well as with any operation that may take place later on the docids & scores. 
The PIT here is closed along with the request when we have the final results to return to the user  from the `FetchLookupFieldsPhase`. 

The main issue here with not having a PIT possibly just within the query phase, is that some retrievers might need additional data (e.g. the semantic reranker) so this whole rewriting is not restricted just to the query phase. Additionally, we could (potentially) have a semantic reranker on top of an rrf retriever, so the idea was that the PIT should ensure that all parts of the search request will have access to the same underlying readers & docs no matter how they're actually executed. 

Do you see any issues that this might cause under heavy load?

Reply:
> Do you see any issues that this might cause under heavy load?

Yes, PITs aren't free. Handling many thousands per second (effectively, every rrf query is now a PIT query), is gonna cost.

https://www.elastic.co/guide/en/elasticsearch/reference/current/point-in-time-api.html#point-in-time-keep-alive

This will add negligible heap overhead, open files, etc. especially if the index is mutating during the search (which is the whole reason for the PIT actually...).

My concern is this:

 - We obviously thought it was "OK" to do RRF previously without opening a PIT
 - Additionally, we thought that it was OK to do the inference rerank without a PIT

Now, whats changed?

Has this behavior always been wrong?

Reply:
Here are a few clarifications. Technically, every request uses a PIT. This PIT is currently created during the first phase of the search (DFS or query phase) and is used consistently in all subsequent phases. The main difference in this PR is that we need to reference this PIT to execute additional requests that are independent of the normal search phases. We achieve this by explicitly creating a PIT instead of waiting for the first phase to create it. Therefore, there are no additional resources used by this change. The only overhead is the extra communication with the shards, as creating the PIT is usually done as part of other work.

Reply:
Thank you for the clarification. One issue I see is `allow_partial_search_results`. I am pretty sure PIT's just fail every time when there is a missing shard instead of bypassing it like a normal search request (which defaults `allow_partial_search_results: true`).